src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1872,12 +1872,14 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
             BoundNode highestBoundNode,
             BoundNode boundNodeForSyntacticParent)
         {
-            if (lowestBoundNode is BoundPattern pattern)
+            BoundPattern pattern = lowestBoundNode as BoundPattern ?? highestBoundNode as BoundPattern ?? (highestBoundNode is BoundSubpattern sp ? sp.Pattern : null);
```

Comment:
Is there a case where `lowestBoundNode` is a BoundSubpattern, or is that case never useful in practice (already handled by other cases)?

Reply:
I think the bound pattern would always be lower than a bound subpattern, as the API rejects a subpattern syntax as input.